openupgrade_scripts/scripts/payment/15.0.2.0/pre-migration.py

```diff
@@ -0,0 +1,76 @@
+from openupgradelib import openupgrade
+
+
+def copy_fields(env):
+    openupgrade.copy_columns(
+        env.cr,
+        {
+            "payment_acquirer": [
+                ("journal_id", None, None),
+            ],
+        },
+    )
+
+
+def create_account_payment_source_payment_id(env):
+    # Create the column first to avoid related
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE account_payment
+        ADD COLUMN IF NOT EXISTS source_payment_id int""",
+    )
+
+
+def fast_fill_payment_token_name(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE payment_token
+        SET name = '/'
+        WHERE name IS NULL""",
+    )
+
+
+def fast_fill_payment_transaction_partner_id(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE payment_transaction pt
+        SET partner_id = (SELECT ap.partner_id
+            FROM account_payment ap
+            WHERE pt.partner_id IS NULL AND
+            ap.payment_transaction_id = pt.id
+            LIMIT 1)""",
```

Comment:
@tranngocson1996 quan hệ là Many2one mà em, sao phải Limit làm gì nhỉ?
```suggestion
        UPDATE payment_transaction pt
        SET partner_id = ap.partner_id
        FROM account_payment ap
        WHERE ap.partner_id IS NOT NULL AND pt.partner_id IS NULL AND ap.payment_transaction_id = pt.id
        """, 
```

Reply:
@royleviindoo anh ơi chỗ này vẫn còn `limit 1` chứ anh nhỉ. Vì tìm như kia có thể ra nhiều payment và mỗi payment lại có 1 partner khác nhau

Reply:
@tranngocson1996 Quan hệ many2one mà, 1 payment.transaction chỉ có 1 payment mà, sao ra nhiều được?